src/main/java/us/kbase/userprofile/MongoController.java

```diff
@@ -138,55 +124,56 @@ public List<User> filterUsers(String filter) {
 					}
 				}
 			}
-			if(d.containsField("thumbnail")) {
+			if(d.containsKey("thumbnail")) {
 				if(d.get("thumbnail")!=null) {
 					u.setThumbnail(d.get("thumbnail").toString());
 				}
 			}
 			
 			// only add if it was found
 			if(add) {
-				//System.out.println(u);
 				users.add(u);
 			}
 		}
 		return users;
 	}
 	
 	private void ensureIndex() {
-		DBObject userUnique = new BasicDBObject("user.username",1);
-		//DBObject userText = new BasicDBObject("user.username","text");
-		DBObject unique = new BasicDBObject("unique",1);
-		//DBObject nameText = new BasicDBObject("user.realname","text");
-		profiles.createIndex(userUnique, unique);
-		//profiles.ensureIndex(userText);
-		//profiles.ensureIndex(nameText);
+		Document userUnique = new Document("user.username",1);
+		IndexOptions uniqueOptions = new IndexOptions().unique(true);
+		profiles.createIndex(userUnique, uniqueOptions);
 	}
-	
-	
+
 	public boolean exists(String username) {
-		DBObject query = new BasicDBObject("user.username",username);
-		if(profiles.findOne(query, new BasicDBObject())==null) {
-			return false;
-		}
-		return true;
+		return findProfileByUsername(username) != null;
+    }
+
+	private Document findProfileByUsername(String username) {
```

Comment:
There's only one method that calls this so you might as well inline

Reply:
👍